source/common/upstream/upstream_impl.h

```diff
@@ -48,13 +50,16 @@ class HealthCheckHostMonitorNullImpl : public HealthCheckHostMonitor {
 class HostDescriptionImpl : virtual public HostDescription {
 public:
   HostDescriptionImpl(ClusterInfoConstSharedPtr cluster, const std::string& hostname,
-                      Network::Address::InstanceConstSharedPtr dest_address, bool canary,
-                      const std::string& zone)
-      : cluster_(cluster), hostname_(hostname), address_(dest_address), canary_(canary),
+                      Network::Address::InstanceConstSharedPtr dest_address,
+                      const HostMetadata& metadata, const std::string& zone)
```

Comment:
should not zone be part of metadata as well?

Reply:
Zone is first class in the EDS API, so it isn't in the Metadata object. If we're building up some additional HostMetadata object, we can put zone in there at config ingestion time, but, see my above comment on HostMetadata.